omod-2.4/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/search/openmrs2_4/OrderSearchHandler2_4.java

```diff
@@ -0,0 +1,223 @@
+	
+	public static final String REQUEST_PARAM_EXCLUDE_CANCELED_AND_EXPIRED = "excludeCanceledAndExpired";
+	
+	public static final String REQUEST_PARAM_INCLUDE_VOIDED = "includeVoided";
+	
+	@Autowired
+	@Qualifier("patientService")
+	PatientService patientService;
+	
+	@Autowired
+	@Qualifier("orderService")
+	OrderService orderService;
+	
+	@Autowired
+	@Qualifier("conceptService")
+	ConceptService conceptService;
+	
+	SearchQuery searchQuery = new SearchQuery.Builder("Allows you to search for orders, it matches on "
+	        + "patient, care setting, concepts (comma delimited), order types (comma delimited), "
+	        + "date activated (before or after), voided flag")
+	        .withOptionalParameters(REQUEST_PARAM_PATIENT,
+	            REQUEST_PARAM_CARE_SETTING,
+	            REQUEST_PARAM_CONCEPTS,
+	            REQUEST_PARAM_ORDER_TYPES,
+	            REQUEST_PARAM_ACTIVATED_ON_OR_BEFORE_DATE,
+	            REQUEST_PARAM_ACTIVATED_ON_OR_AFTER_DATE,
+	            REQUEST_PARAM_IS_STOPPED,
+	            REQUEST_PARAM_AUTO_EXPIRE_ON_OR_BEFORE_DATE,
+	            REQUEST_PARAM_CANCELED_OR_AUTO_EXPIRE_ON_OR_BEFORE_DATE,
+	            REQUEST_PARAM_ACTION,
+	            REQUEST_PARAM_FULFILLER_STATUS,
+	            REQUEST_PARAM_EXCLUDE_CANCELED_AND_EXPIRED,
+	            REQUEST_PARAM_INCLUDE_VOIDED).build();
+	
+	private final SearchConfig searchConfig = new SearchConfig("default", RestConstants.VERSION_1
+	        + "/order", Arrays.asList("2.4.*"), searchQuery);
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.SearchHandler#getSearchConfig()
+	 */
+	@Override
+	public SearchConfig getSearchConfig() {
+		return searchConfig;
+	}
+	
+	/**
+	 * @see SearchHandler#search(RequestContext)
+	 */
+	@Override
+	public PageableResult search(RequestContext context) throws ResponseException {
+		// get input parameters
+		String patientUuid = context.getParameter("patient");
+		String careSettingUuid = context.getParameter("careSetting");
+		String conceptUuids = context.getParameter("concepts");
+		String orderTypeUuids = context.getParameter("orderTypes");
+		String activatedOnOrBeforeDateStr = context.getParameter("activatedOnOrBeforeDate");
+		String activatedOnOrAfterDateStr = context.getParameter("activatedOnOrAfterDate");
+		String isStoppedStr = context.getParameter(REQUEST_PARAM_IS_STOPPED);
+		String autoExpireOnOrBeforeDateStr = context.getParameter(REQUEST_PARAM_AUTO_EXPIRE_ON_OR_BEFORE_DATE);
+		String canceledOrExpiredOnOrBeforeDateStr = context
+		        .getParameter(REQUEST_PARAM_CANCELED_OR_AUTO_EXPIRE_ON_OR_BEFORE_DATE);
+		String actionStr = context.getParameter(REQUEST_PARAM_ACTION);
+		String fulfillerStatusStr = context.getParameter(REQUEST_PARAM_FULFILLER_STATUS);
+		String excludeCanceledAndExpiredStr = context.getParameter(REQUEST_PARAM_EXCLUDE_CANCELED_AND_EXPIRED);
+		String includeVoidedStr = context.getParameter("includeVoided");
```

Comment:
Nit, but any reason you aren't using the constant here as well... ?

Reply:
Thanks @mogoodrich . I have replaced those strings with constants.